rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp

```diff
@@ -138,7 +138,24 @@ namespace rsx
 				glPixelStorei(GL_UNPACK_SWAP_BYTES, GL_TRUE);
 				glPixelStorei(GL_UNPACK_ROW_LENGTH, pitch / 2);
 
-				// TODO: texture swizzling
+				if (is_swizzled)
+				{
+					u16 *src, *dst;
+					u16 height = tex.height();
+					u16 width = tex.width();
+
+					unswizzledPixels = (u8*)malloc(width * height * 2);
```

Comment:
I chose to implement it that way to retain consistency with the rest of the file, retaining variable types and naming with other areas where similar functionality has been implemented. However, I  agree that the unique_ptr would've been a more elegant solution.